cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/ProgramRuntimeLifecycleHttpHandler.java

```diff
@@ -0,0 +1,481 @@
+      @PathParam("app-id") String appId,
+      @PathParam("service-id") String serviceId) throws Exception {
+    try {
+      ProgramId programId = new ApplicationId(namespaceId, appId,
+          getLatestAppVersion(new NamespaceId(namespaceId), appId))
+          .program(ProgramType.SERVICE, serviceId);
+      Store.ensureProgramExists(programId, store.getApplication(programId.getParent()));
+
+      int instances = getInstances(request);
+      runtimeLifecycleService.setInstances(programId, instances);
+      responder.sendStatus(HttpResponseStatus.OK);
+    } catch (SecurityException e) {
+      responder.sendStatus(HttpResponseStatus.UNAUTHORIZED);
+    } catch (Throwable throwable) {
+      if (respondIfElementNotFound(throwable, responder)) {
+        return;
+      }
+      throw throwable;
+    }
+  }
+
+  /**
+   * Returns number of instances of a worker.
+   */
+  @GET
+  @Path("/apps/{app-id}/workers/{worker-id}/instances")
+  public void getWorkerInstances(HttpRequest request, HttpResponder responder,
+      @PathParam("namespace-id") String namespaceId,
+      @PathParam("app-id") String appId,
+      @PathParam("worker-id") String workerId) throws Exception {
+    try {
+      ProgramId programId = validateAndGetNamespace(namespaceId)
+          .app(appId, getLatestAppVersion(new NamespaceId(namespaceId), appId))
+          .worker(workerId);
+      lifecycleService.ensureProgramExists(programId);
+      int count = store.getWorkerInstances(programId);
+      responder.sendJson(HttpResponseStatus.OK, GSON.toJson(new Instances(count)));
+    } catch (SecurityException e) {
+      responder.sendStatus(HttpResponseStatus.UNAUTHORIZED);
+    } catch (Throwable e) {
+      if (respondIfElementNotFound(e, responder)) {
+        return;
+      }
+      throw e;
+    }
+  }
+
+  /**
+   * Sets the number of instances of a worker.
+   */
+  @PUT
+  @Path("/apps/{app-id}/workers/{worker-id}/instances")
+  @AuditPolicy(AuditDetail.REQUEST_BODY)
+  public void setWorkerInstances(FullHttpRequest request, HttpResponder responder,
+      @PathParam("namespace-id") String namespaceId,
+      @PathParam("app-id") String appId,
+      @PathParam("worker-id") String workerId) throws Exception {
+    int instances = getInstances(request);
+    try {
+      runtimeLifecycleService.setInstances(new ApplicationId(namespaceId, appId,
+          getLatestAppVersion(new NamespaceId(namespaceId), appId))
+          .program(ProgramType.WORKER, workerId), instances);
+      responder.sendStatus(HttpResponseStatus.OK);
+    } catch (SecurityException e) {
+      responder.sendStatus(HttpResponseStatus.UNAUTHORIZED);
+    } catch (Throwable e) {
+      if (respondIfElementNotFound(e, responder)) {
+        return;
+      }
+      throw e;
+    }
+  }
+
+  @GET
+  @Path("/apps/{app-id}/{program-category}/{program-id}/live-info")
+  @SuppressWarnings("unused")
+  public void liveInfo(HttpRequest request, HttpResponder responder,
```

Comment:
Please add javadoc

Reply:
Done.

Added java docs for all new classes and public functions.